uniprotkb-rest/src/test/java/uk/ac/ebi/uniprot/api/uniprotkb/controller/UniprotKBSearchControllerIT.java

```diff
@@ -759,10 +352,417 @@ public void searchForDeletedInactiveEntriesReturnItself() throws Exception {
                 .andExpect(jsonPath("$.results.*.inactiveReason.inactiveReasonType", contains("DELETED")));
     }
 
-    private String saveEntry() {
-        UniProtEntry entry = UniProtEntryMocker.create(UniProtEntryMocker.Type.SP);
-        String acc = entry.getPrimaryAccession().getValue();
-        storeManager.save(DataStoreManager.StoreType.UNIPROT, entry);
-        return acc;
+    @Override
+    protected void cleanEntries() {
+        storeManager.cleanSolr(DataStoreManager.StoreType.UNIPROT);
     }
+
+    @Override
+    protected MockMvc getMockMvc() {
+        return mockMvc;
+    }
+
+    @Override
+    protected String getSearchRequestPath() {
+        return SEARCH_RESOURCE;
+    }
+
+    @Override
+    protected int getDefaultPageSize() {
+        return 25;
+    }
+
+    @Override
+    protected List<SearchField> getAllSearchFields() {
+        return Arrays.asList(UniProtField.Search.values());
+    }
+
+    @Override
+    protected String getFieldValueForValidatedField(SearchField searchField) {
+        UniProtField.Search search = UniProtField.Search.valueOf(searchField.getName());
+        log.debug("FIELD_NAME: ", search.getName());
+        String value = "";
+        if (searchField.getName().startsWith("ftlen_")) {
+            value = "[* TO *]";
+        }
+        switch (search) {
+            case accession:
+            case accession_id:
+                value = "P21802";
+                break;
+            case organism_id:
+            case host_id:
+            case taxonomy_id:
+                value = "9606";
+                break;
+            case modified:
+            case created:
+            case sequence_modified:
+            case lit_pubdate:
+            case length:
+            case mass:
+                value = "[* TO *]";
+                break;
+            case proteome:
+                value = "UP000000000";
+                break;
+            case annotation_score:
+                value = "5";
+                break;
+        }
```

Comment:
should we have a default case, which throws some kind of runtimeexception to show that something is wrong?

Reply:
I can not use the default here... because if the result of this method is empty I do extract things in the abstract layer... This is that calls this method:

    private String getFieldValueForField(SearchField searchField) {
        String value = getFieldValueForValidatedField(searchField);
        if (value.isEmpty()) {
            if (searchField.hasValidValue("*")) {
                value = "*";
            } else if (searchField.hasValidValue("true")) {
                value = "true";
            }
        }
        return value;
    }